src/librustc_typeck/check/mod.rs

```diff
@@ -684,7 +684,9 @@ impl Inherited<'a, 'tcx> {
             deferred_generator_interiors: RefCell::new(Vec::new()),
             opaque_types: RefCell::new(Default::default()),
             opaque_types_vars: RefCell::new(Default::default()),
-            implicit_region_bound: None,
+            // Make sure that we can always assume `T: ReEmpty(U0)` for any
+            // type parameter `T`.
+            implicit_region_bound: Some(tcx.lifetimes.re_root_empty),
```

Comment:
I suspect this field could be made non-optional if we use empty here

Reply:
I'm a bit confused as to how this winds up fixing any bug -- reading the code that is using this field, it looks like it tries to make `'empty` an implicit bound:

https://github.com/rust-lang/rust/blob/71384101ea3b030b80f7def80a37f67e148518b0/src/librustc_infer/infer/outlives/verify.rs#L73-L99

maybe `VerifyBound::IsEmpty` can be removed after this change, even?

Reply:
The issue is that it's possible that we have `T: ReEmpty(Un)` as an implied bound from a closure, which causes us to not generate `VerifyBound::IsEmpty`.